### DISCUSSION TITLE
Minimal version

### DISCUSSION THREAD

**Discussion by ajrty33** [tim-hub/obsidian-bible-reference#72] [Ideas]
I have been modifying src/VerseSuggesting.ts to create a minimal version. When taking notes in class with a lot of Scripture references, it's nice to have a bare bones verse without anything else. I would love to create a pull request for this, but creating the settings to make this an option is above my head.
![image](https://user-images.githubusercontent.com/1326379/210857873-098c45b1-1be7-4ac4-9946-6a5408d037b5.png)

**Comment by aknisly** [comment]
This is almost exactly what I was wishing for. You inspired me to dig into it myself, and your tip about the place to tweak led me to success. I've matched yours, but I'd still like to remove the newline before the reference, but I can't find it.
If anybody can help me with that, I'd be grateful.
I agree that this option would be a great feature to add, and I, too, am out of my league to add it.
I've attached my modified _main.js_ in case anyone else would like to use it. It's located in _.obsidian/plugins/obsidian-bible-reference_. I've added comments to show what I've tweaked. (NOTE: I had to add '.txt' to the file name to upload it; remove that when you download it.) 
[main.js.txt](https://github.com/tim-hub/obsidian-bible-reference/files/10759493/main.js.txt)

**Reply by ajrty33** [reply]
https://github.com/tim-hub/obsidian-bible-reference/blob/a481d3595dd617c9f25fd9603e766541bc95a3d9/src/VerseSuggesting.ts#L61
I haven't updated the plugin, but it was just this:
```
    return [
      /*head,*/
      this.text,
      bottom
    ].join("\n");
```
```
    return [
      /*head,*/
      this.text,
      bottom
    ].join(" "); //space here
```
It looks like you deleted the new line on line 1017 of your file though...
`    return [head, this.text, bottom].join("");`
I plan to update and look at it this weekend

**Comment by ajrty33** [comment]
oh.... Line 1011 and 1012 in your text file... You just need to delete the actual new line in the quotes...
```
      bottom += ` 
 ${this.getVerseReference()}`;
```
```
      bottom += `${this.getVerseReference()}`;
```

**Reply by aknisly** [reply]
Wow...that was too obvious, except I didn't understand what was going on. Thanks much!

**Comment by tim-hub** [comment]
This is a good point, `simple is better`

I am thinking of maybe add the option of having `bare bone` verse block.

and use it as a default settings

**Comment by tim-hub** [comment]
Will add this to todo column

**Comment by aknisly** [comment]
In case others find it helpful, I've updated main.js in the current version (1.7.3) to tweak the formatting as per my earlier efforts. Thanks again to @tim-hub for your work on this plugin!

[main.js.txt](https://github.com/tim-hub/obsidian-bible-reference/files/12059932/main.js.txt)

**Reply by tim-hub** [reply]
Thanks @aknisly for the update and sponsoring. 

Do you mind share they change you made in `main.js`? The changes only?, not the whole file? 


Quite hard to read a build JS version file

**Reply by aknisly** [reply]
Sorry, @tim-hub! I missed your request. Here is the patch file generated by Meld for the changes I made to the main.js for v. 2.3.7
[patch file (changes to main.js).txt](https://github.com/tim-hub/obsidian-bible-reference/files/13573240/patch.file.changes.to.main.js.txt)

**Comment by aknisly** [comment]
I'm not sure if this should be a new post or not, but I just thought of another possible enhancement. If I'm inserting a single verse, the verse number is basically superfluous. Would it be helpful to add an option to turn off verse numbers when inserting a single verse? Just another idea....

**Comment by tim-hub** [comment]
@ajrty33 Would you mind trying the new version 2.4.0

With the new settings, it will be able to get a minimal version output now.

https://antioch.tech/obsidain-bible-reference-plugin-2-4-0-changelog/


I will close this discussion for now, but feel free to raise concern and issues

**Comment by aknisly** [comment]
This is a nice improvement! Thanks again for your work on this.
For anybody else wishing to include the reference, but strip it down a bit further, I removed the link from the reference by making this change in the main.js as described above: 
> return \` -- ~\[~${this.bibleProvider.BibleReferenceHead} - ${this.bibleVersion.toUpperCase()}~\](${this.bibleProvider.VerseLinkURL})~\`;
 
Note the "[" that has been removed right before the first "$". (Sorry, I couldn't figure out how to escape the backtick; hence,the syntax for a quote instead of code.) 

This renders the reference thus: 
`-- Exodus 24:8 - ESV`
 
I also like this without the quote syntax and newlines, so I tweaked the main.js as per the patch file attached.
[patch file (changes to main.js--v. 2.4.2).txt](https://github.com/tim-hub/obsidian-bible-reference/files/14530794/patch.file.changes.to.main.js--v.2.4.2.txt)
FWIW....
